topdown/http_test.go

```diff
@@ -340,3 +344,213 @@ func getTestServer() (baseURL string, teardownFn func()) {
 
 	return ts.URL, ts.Close
 }
+
+func getTLSTestServer() (ts *httptest.Server) {
+	mux := http.NewServeMux()
+	ts = httptest.NewUnstartedServer(mux)
+
+	mux.HandleFunc("/test", func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(http.StatusOK)
```

Comment:
I suppose to fully round-trip this, we could have the handler return some information about the used client cert? There's a `TLS` field on [`req *http.Request`](https://golang.org/pkg/net/http/#Request) that could come in handy.

Reply:
@repenno this seems like an important attribute of these tests. Can we have at least one success case very that the information in the client certificate is correct?

Reply:
Done, I created a test that reflects back the certificate common name